### PR TITLE
Upgrade to latest React Native

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-14625
affects: Student, Teacher
release note: none
[run-nightly]

### CODE REVIEW

**Comment by vanwagonet** [comment]
test plan:
Try some push notifications, there is one in the JIRA issue. You can also check to see that routing to /courses, /to-do, /inbox, & some other non-main screen routes work too.

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/168h4v9) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F168h4v9)
teacher: [Link](https://rebrand.ly/39zxz88) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F39zxz88)
parent: [Link](https://rebrand.ly/y4wnijv) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fy4wnijv)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* rn/Teacher/src/common/components/RefreshableList.js
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


<details><summary>:x: Yarn lint</summary>


    
    /Users/vagrant/git/rn/Teacher/index.ios.js
      23:3  error  'AppRegistry' is defined but never used  no-unused-vars
    
    /Users/vagrant/git/rn/Teacher/src/common/components/RefreshableList.js
      19:1  error  More than 1 blank line not allowed  no-multiple-empty-lines
    
    ✖ 2 problems (2 errors, 0 warnings)
      1 error and 0 warnings potentially fixable with the `--fix` option.
    
</details>


[MBL-14625](https://instructure.atlassian.net/browse/MBL-14625)

#### Affected Apps: Student, Teacher
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by cobbal** [comment]
QA+1

File: rn/Teacher/src/modules/submissions/SubmissionGraph.js [hunk 1]
```diff
@@ -30,10 +30,51 @@ import {
...
+    const easeVal = this.circInOut(percentage)
+    const value = start + (end - start) * easeVal
+
+    this.setState({ value })
+  }
+
+  circInOut (t) {
+    return ((t *= 2) < 1)
+      ? -0.5 * (Math.sqrt(1 - t * t) - 1)
+      : 0.5 * (Math.sqrt(1 - (t -= 2) * t) + 1)
```
Comment (armstrongnate) [inline rn/Teacher/src/modules/submissions/SubmissionGraph.js:75]: image

**Pull request merged**